recipes-app/iot2050-firmware-update/files/iot2050-firmware-update

```diff
@@ -9,6 +9,30 @@
 # This file is subject to the terms and conditions of the MIT License.  See
 # COPYING.MIT file in the top-level directory.
 #
+# To use this tool, a <firmware-update-package>.tar.xz and its sha256 checksum file
+# <firmware-update-package>.tar.xz.sha256 are needed.
+#
+# The <firmware-update-package>.tar.xz contains below files:
+#   - firmware.bin: The firmware to update.
+#   - firmware.bin.sha256: The sha256 checksum of the firmware.
+#   - update.conf.json: The update criteria. Currently only the OS image version
+#       is checked.
```

Comment:
What about providing multiple firmware binaries for different targets? Should the user get a single tarball, and the tool will pick the fitting one? Or will we distribute multiple tarball, and the tool will only reject those that do not fit?

Reply:
There will be different binaries for different board, and also different sha256 checksums. All of these artifacts will be put into one tarball, and the tool will pick one depends on the physical board model.

This PR is only targeting next image release of PG1, I will work on the next version of this tool after transition kernel image released to provide these 'enhanced' functionalties.

Reply:
We will have a chicken-egg issue here if we do not account for the PG2 case already: How do you want to ship a new firmware version for all hardware variants later on when this updater cannot match to PG1?

Also, checksumming, inside and outside of the taball, seems total overkill now, given that has validation mechanisms.

Reply:
We will have an new example release targeting PG2 board, and these change could be live in that version.

Regarding checksum, I think we will only provide one checksum of the tarball on the download page, and let user to do the verifying by themselves. Then it's reasonale to remove all the checksumming from this tool.

Reply:
Let's better do this properly now, rather than regretting having had all measures in place already, ie. already add hw detection, description and binary selection today. Should be simple enough (via fw-env or /sys/firmware/devicetree).

Reply:
To make it more concrete what I'm concerned about: Customer has the upcoming image installed on a PG1 device. We prepared a firmware update for all devices, but our image tool in that version you propose now will either needlessly stumble over that new format ("best" case) or even pick the wrong image from the tarball (worst case) and brick the device.